packages/issuer-token-endpoint/lib/IssuerTokenEndpoint.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  AccessTokenResponse,
+  Alg,
+  CNonceState,
+  CredentialOfferState,
+  getNumberOrUndefined,
+  IStateManager,
+  Jwt,
+  JWTSignerCallback,
+  Typ,
+} from '@sphereon/openid4vci-common'
+import express, { NextFunction, Request, Response, Router } from 'express'
+import { v4 } from 'uuid'
+
+const router = express.Router()
+
+interface ITokenEndpointOpts {
+  tokenPath?: string
+  interval?: number
+  cNonceExpiresIn?: number
+  tokenExpiresIn?: number
+  stateManager?: IStateManager<CredentialOfferState>
+  nonceStateManager?: IStateManager<CNonceState>
+  jwtSignerCallback?: JWTSignerCallback
+}
+
+let tokenPath: string
+let interval: number
+let cNonceExpiresIn: number
+let tokenExpiresIn: number
+let stateManager: IStateManager<CredentialOfferState>
+let nonceStateManager: IStateManager<CNonceState>
+let jwtSignerCallback: JWTSignerCallback
+
+export const tokenRequestEndpoint = (opts?: ITokenEndpointOpts): Router => {
+  tokenPath = (opts?.tokenPath ? opts.tokenPath : (process.env.TOKEN_PATH as string)) ?? '/token'
```

Comment:
Interesting you are mixing a ternary with nullish coalescing. Why not use nullish coalescence everywhere then?
````typescript
opts?.tokenPath ?? process.env.TOKEN_PATH as string ?? '/token'
````

Reply:
I'm rocker but never got to know Elvis' work. :laughing: Now I see it's possible to chain them.